app/models/concerns/twitter_user/builder.rb

```diff
@@ -0,0 +1,114 @@
+require 'active_support/concern'
+
+module Concerns::TwitterUser::Builder
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def build_by_user(user, attrs = {})
+      build_relation = attrs.has_key?(:build_relation) ? attrs.delete(:build_relation) : false
+      tu = new(attrs) do |tu|
+        tu.uid = user.id
+        tu.screen_name = user.screen_name
+        tu.user_info = user.slice(*TwitterUser::PROFILE_SAVE_KEYS).to_json # TODO check the type of keys and values
+      end
+      tu.build_relations if build_relation
+      tu
+    end
+
+    def build_by_client(client, user, attrs = {})
+      build_by_user(client.user(user), attrs.merge(client: client))
+    end
+
+    def build(client, user, option = {})
+      build_by_client(client, user, option)
+    end
+
+  end
+
+  included do
+    def build_relations
```

Comment:
__[rubocop]__
* Assignment Branch Condition size for build_relations is too high. [78.32/15] [:ref](http://c2.com/cgi/wiki?AbcMetric)
* Method has too many lines. [71/10] [:ref](https://github.com/bbatsov/ruby-style-guide#short-methods)